RadixWallet/Features/AssetsFeature/Components/HelperViews/ResourceBalance/ResourceBalanceView.swift

```diff
@@ -543,78 +613,229 @@ extension ResourceBalanceView {
 	}
 
 	struct AmountView: View {
-		@Environment(\.resourceBalanceHideFiatValue) var resourceBalanceHideFiatValue
-		let amount: ResourceBalance.Amount?
+		let amount: KnownResourceBalance.Amount?
 		let fallback: String?
-		let compact: Bool
+		let appearance: Appearance
+		let symbol: Loadable<String?>?
 
-		init(amount: ResourceBalance.Amount?, fallback: String? = nil, compact: Bool) {
+		enum Appearance: Sendable, Equatable {
+			case standard
+			case compact
+			case large
+		}
+
+		init(
+			amount: KnownResourceBalance.Amount?,
+			fallback: String? = nil,
+			appearance: Appearance,
+			symbol: Loadable<String?>? = nil
+		) {
 			self.amount = amount
 			self.fallback = fallback
-			self.compact = compact
+			self.appearance = appearance
+			self.symbol = symbol
 		}
 
 		var body: some View {
 			if let amount {
-				core(amount: amount, compact: compact)
+				switch amount.amount {
+				case let .exact(exactAmount):
+					SubAmountView(
+						amount: exactAmount,
+						guaranteed: amount.guaranteed,
+						appearance: appearance,
+						symbol: symbol
+					)
+				case let .atLeast(exactAmount):
+					SubAmountView(
+						title: L10n.InteractionReview.atLeast,
+						amount: exactAmount,
+						guaranteed: amount.guaranteed,
+						appearance: appearance,
+						symbol: symbol
+					)
+				case let .atMost(exactAmount):
+					SubAmountView(
+						title: L10n.InteractionReview.noMoreThan,
+						amount: exactAmount,
+						guaranteed: amount.guaranteed,
+						appearance: appearance,
+						symbol: symbol
+					)
+				case let .between(minAmount, maxAmount):
+					VStack(alignment: alignment, spacing: .small3) {
+						SubAmountView(
+							title: L10n.InteractionReview.atLeast,
+							amount: minAmount,
+							appearance: appearance,
+							symbol: symbol
+						)
+						SubAmountView(
+							title: L10n.InteractionReview.noMoreThan,
+							amount: maxAmount,
+							appearance: appearance,
+							symbol: symbol
+						)
+					}
+				case .unknown:
+					EmptyView()
+				}
 			} else if let fallback {
 				Text(fallback)
 					.textStyle(amountTextStyle)
 					.foregroundColor(.app.gray2)
 			}
 		}
 
-		@ViewBuilder
-		private func core(amount: ResourceBalance.Amount, compact: Bool) -> some View {
-			if compact {
+		private var amountTextStyle: TextStyle {
+			switch appearance {
+			case .standard, .large:
+				.secondaryHeader
+			case .compact:
+				.body1HighImportance
+			}
+		}
+
+		private var alignment: HorizontalAlignment {
+			switch appearance {
+			case .standard, .compact:
+				.trailing
+			case .large:
+				.center
+			}
+		}
+	}
+
+	struct SubAmountView: View {
```

Comment:
What does `Sub` mean :)?

Reply:
`SubAmountView` indicates it's a smaller part or subset of the main `AmountView`. I could rename it to `AmountItemView` if that’s clearer.